src/FESpaces/FESpacesWithDofPotentiallyRemoved.jl

```diff
@@ -0,0 +1,185 @@
+"""
+    struct FESpaceWithDofPotentiallyRemoved{CS,RD} <: SingleFieldFESpace
+      space::SingleFieldFESpace
+      constraint_style::Val{CS}
+      remove_dof::Val{RD}
+      dof_to_remove::Int
+    end
+"""
+struct FESpaceWithDofPotentiallyRemoved{CS,RD} <: SingleFieldFESpace
+  space::SingleFieldFESpace
+  constraint_style::Val{CS}
+  remove_dof::Val{RD}
+  dof_to_remove::Int
```

Comment:
I am not sure you need `constraint_style` and `remove_dof`, already in the param.

Reply:
Proposal (feedback welcomed):

```
abstract type ConstantStrategy end;
type FixConstant           <: ConstantStrategy
type DoNotFixConstant <: ConstantStrategy
```

With `constraint_style` I need help. May be @fverdugo can help here? What does `True/False` mean for the `constraint_style` of FE spaces? On the other hand, I guess this is a cross-cutting change, right?

Reply:
> With constraint_style I need help. May be @fverdugo can help here? What does True/False mean for the constraint_style of FE spaces? On the other hand, I guess this is a cross-cutting change, right?

`constraint_style(V) == Val(false)` for un-constrained spaces

Reply:
Done (partially) in 0da971d918e87711098e3a32e14036c9741a4cde. In particular, I would prefer to transform `constraint_style` into struct-like traits in a different PR. I expect that such implies a number of changes that deserve its separate PR.